src/daft-logical-plan/src/optimization/rules/simplify_expressions.rs

```diff
@@ -0,0 +1,512 @@
+use std::sync::Arc;
+
+use common_error::DaftResult;
+use common_scan_info::PhysicalScanInfo;
+use common_treenode::{Transformed, TreeNode};
+use daft_core::prelude::SchemaRef;
+use daft_dsl::{lit, null_lit, Expr, ExprRef, LiteralValue, Operator};
+use daft_schema::dtype::DataType;
+
+use super::OptimizerRule;
+use crate::LogicalPlan;
+
+/// Optimization rule for simplifying expressions
+#[derive(Default, Debug)]
+pub struct SimplifyExpressionsRule {}
+
+impl SimplifyExpressionsRule {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl OptimizerRule for SimplifyExpressionsRule {
+    fn try_optimize(&self, plan: Arc<LogicalPlan>) -> DaftResult<Transformed<Arc<LogicalPlan>>> {
```

Comment:
You should probably use `plan.transform` here so that every plan node and not just the root is optimized